src/components/project/project.rules.ts

```diff
@@ -1010,14 +1032,28 @@ export class ProjectRules {
   }
 
   /** A list of the project's previous steps ordered most recent to furthest in the past */
-  private async getPreviousSteps(id: ID): Promise<ProjectStep[]> {
+  private async getPreviousSteps(
+    id: ID,
+    changeset?: ID
+  ): Promise<ProjectStep[]> {
     const result = await this.db
       .query()
       .match([
+        ...(changeset
+          ? [
+              node('changeset', 'Changeset', { id: changeset }),
+              relation('in', '', 'changeset', ACTIVE),
+            ]
+          : []),
         node('node', 'Project', { id }),
-        relation('out', '', 'step', INACTIVE),
+        relation('out', '', 'step', changeset ? undefined : INACTIVE),
         node('prop'),
       ])
+      .apply((q) =>
+        changeset
+          ? q.raw('WHERE NOT (changeset)-[:changeset {active:true}]->(prop)')
```

Comment:
`getPreviousSteps` is updated being changeset aware. In changeset mode, it gets all(`ACTIVE` & `INACTIVE`) step props except the current active changeset step.